platform/darwin/src/MGLOfflineStorage.mm

```diff
@@ -299,7 +216,112 @@ - (void)observeValueForKeyPath:(NSString *)keyPath ofObject:(id)object change:(N
     }
 }
 
-#pragma mark Offline merge methods
+#pragma mark Database management methods
+
+- (NSString *)databasePath {
+    return self.databaseURL.path;
+}
+
+- (NSURL *)databaseURL {
+    if (!_databaseURL) {
+        NSString *customPath = [NSBundle.mainBundle objectForInfoDictionaryKey:MGLOfflineStorageDatabasePathInfoDictionaryKey];
+        if ([customPath isKindOfClass:[NSString class]]) {
+            _databaseURL = [NSURL fileURLWithPath:customPath.stringByStandardizingPath isDirectory:NO relativeToURL:NSBundle.mainBundle.resourceURL];
+            
+            NSURL *directoryURL = _databaseURL.URLByDeletingLastPathComponent;
+            [NSFileManager.defaultManager createDirectoryAtURL:directoryURL
+                                   withIntermediateDirectories:YES
+                                                    attributes:nil
+                                                         error:nil];
```

Comment:
As with the default code path in `-defaultDatabaseURLIncludingSubdirectory:`, this custom path code path results in the containing directory being created if it doesn’t already exist. However, this code path doesn’t exclude the containing directory from iCloud backups. Info.plist could’ve specified a path in the same folder as other resources. Besides, the developer might decide that the database won’t be so large and that they don’t want to regenerate the database in case the application gets restored from a backup.